test/steps/configuration_steps.go

```diff
@@ -1,19 +1,40 @@
 package steps
 
 import (
+	"fmt"
+	"os/exec"
 	"strconv"
 
 	"github.com/DATA-DOG/godog"
 	"github.com/pkg/errors"
 )
 
 // ConfigurationSteps defines Cucumber step implementations around configuration.
+// nolint:funlen
 func ConfigurationSteps(suite *godog.Suite, fs *FeatureState) {
+	suite.Step(`^Git Town is no longer configured for this repository$`, func() error {
+		outcome, err := fs.activeScenarioState.gitEnvironment.DeveloperRepo.Run("git", "config", "--local", "--get-regex", "git-town")
+		exitError := err.(*exec.ExitError)
+		if exitError.ExitCode() != 1 {
+			return errors.New("git config should return exit code 1 if no matching configuration found")
+		}
+		if outcome.OutputSanitized() != "" {
+			return errors.Wrapf(err, "expected no local Git Town configuration but got %q", outcome.Output())
```

Comment:
Could the code reach here? Based on the other error message, it doesn't appear so. Maybe should combine the two error messages?

Reply:
I structured the code this way to document that (A) `git config` should return an exit code of 1 here, and (B) `git config` should display no configuration data. The exit code is kind of unexpected, that's why it makes sense to me to document it that way. 

I'm concerned that combining these two aspects leads to an ambiguous check that isn't more clear than the current code:

```go
if exitError.ExitCode() != 1 || outcome.OutputSanitized() != "" {
  return errors.New("unexpected exit code (%d) or configuration found: %q", exitError.ExitCode(), outcome.Output())
}
```

I prefer that the error message specifies exactly what the problem is, instead of giving two possibilities.

It is possible that in an error situation the second one never reaches. But both checks are performed for passing tests.